test/acvp_test.c

```diff
@@ -1357,14 +1361,14 @@ static int rsa_keygen_test(int id)
                                         &d, &d_len)))
         goto err;
 
-    if (!TEST_mem_eq(tst->p1, tst->p1_len, p1, p1_len)
-        || !TEST_mem_eq(tst->p2, tst->p2_len, p2, p2_len)
+    if ((tst->p1 != NULL && !TEST_mem_eq(tst->p1, tst->p1_len, p1, p1_len))
+        || (tst->p2 != NULL && !TEST_mem_eq(tst->p2, tst->p2_len, p2, p2_len))
+        || (tst->q1 != NULL && !TEST_mem_eq(tst->q1, tst->q1_len, q1, q1_len))
+        || (tst->q2 != NULL && !TEST_mem_eq(tst->q2, tst->q2_len, q2, q2_len))
```

Comment:
The NULL checks before TEST_mem_eq are not required and should be removed.

Reply:
The acvp test data that was generated did not contain these fields which is why these checks were added.

Reply:
Err this is not true.. The case I have is test->p1 == NUL test->p1_len == 0, p2 != NULL, p2_len != 0.
These are not equal so it will fail. I dont want it to fail if there is no test data for the field.
(They have changed the outputted fields in newer the CAVP data).